src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+        message.append(buffer);                               //
+
+        if (((y + 1) % 16) == 0)
+        {
+            message += "\n";
+            ShowDebug(message.c_str());
+            message.clear();
+        }
+    }
+
+    if (message.length() > 0)
+    {
+        message += "\n";
+        ShowDebug(message.c_str());
+    }
+}
+
+std::string getHashFromPacket(std::string ip_str, char* data)
+{
+    std::string hash = std::string(data + 12, 16);
+
+    if (authenticatedSessions_[ip_str].find(hash) == authenticatedSessions_[ip_str].end())
+    {
+        return "";
+    }
+    return hash;
+}
+
+// Main menu (Lobby), port 54001
+// A comment on the packets below, defined as macros.
+//   byte 0 - packet size
+//   bytes 4-7 are the packet header "IXFF" (0x49, 0x58, 0x46, 0x46)
+//   byte 8 - The expected message type:
+//     0x03 - positive result
+//     0x04 - error (in the case of an error, a uint16 error code is used at byte 32)
+//     Other
+class view_session : public handler_session
+{
+public:
+    view_session(asio::ssl::stream<asio::ip::tcp::socket> socket)
+    : handler_session(std::move(socket))
+    {
+        DebugSockets("view_session");
+    }
+
+protected:
+    void read_func() override
+    {
+        uint8 code = ref<uint8>(data_, 8);
+
+        std::string sessionHash = getHashFromPacket(ipAddress, data_);
+
+        if (sessionHash == "")
+        {
+            ShowWarning("Session requested without valid sessionHash!");
+            return;
+        }
+        session_t& session = get_authenticated_session(ipAddress, sessionHash);
+        if (!session.view_session)
+        {
+            session.view_session = std::make_shared<view_session>(std::forward<asio::ssl::stream<asio::ip::tcp::socket>>(socket_));
+        }
+        session.view_session->sessionHash = sessionHash;
+
+        DebugSockets(fmt::format("view code: {}", code));
+
+        switch (code)
+        {
+            case 0x07: // 07: "Notifying lobby server of current selections."
+            {
+                auto requestedCharacterID                 = ref<uint32>(data_, 28);
+                char requestedCharacter[PacketNameLength] = {};
+                std::memcpy(&requestedCharacter, data_ + 36, PacketNameLength - 1);
+
+                const char* pfmtQuery = "SELECT accid FROM chars WHERE charid = %u and charname = '%s' LIMIT 1;";
+                auto        sql       = std::make_unique<SqlConnection>();
+
+                uint32 accountID = 0;
+                int32  ret       = sql->Query(pfmtQuery, requestedCharacterID, requestedCharacter);
+                if (ret != SQL_ERROR && sql->NumRows() != 0 && sql->NextRow() == SQL_SUCCESS)
+                {
+                    accountID                    = sql->GetUIntData(0);
+                    session.requestedCharacterID = requestedCharacterID;
+                }
+                else
+                {
+                    // TODO: what's the error code to client here?
```

Comment:
Can we handle this?

Reply:
Couldn't produce this error on retail even if I wanted to. I have no tooling for this, but I could just throw an error at them anyway.

Reply:
This was replaced with closing the socket. It will produce an error.